app/controllers/admin/salesforce_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Admin
+  class SalesforceController < BaseController
+    layout 'admin'
+
+    def show
+    end
+
+    def callback
+      user = SalesforceUser.save_from_omniauth!(env["omniauth.auth"])
+      SalesforceUser.all.reject{|uu| uu.id == user.id}.each(&:destroy)
```

Comment:
I'm ok with it being an ActiveRecord, tho agree it is a little odd to have a table with max one record.  I guess I could also stick this in an `after_create` callback on the AR.  I don't like those callbacks generally, but this is a reasonable case and it only deals with other instances of the AR (and failures there would cause the whole thing to rollback).